src/banking/sheba.rs

```diff
@@ -0,0 +1,356 @@
+use crate::impl_trait_for_string_types;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+lazy_static! {
+    static ref SHEBA_CODE_PATTERN: Regex = Regex::new(r#"^IR[0-9]{24}$"#).unwrap();
+}
+
+#[derive(Debug)]
+pub struct ShebaInfo {
+    nickname: &'static str,
+    name: &'static str,
+    persian_name: &'static str,
+    code: &'static str,
+    account_available: bool,
+}
+
+pub trait ShebaNumber: AsRef<str> {
+    fn is_valid_sheba_code(&self) -> crate::Result<bool> {
```

Comment:
I think it doesn't make sense to return Result here.
You returned Result, because of `iso_7064_mod_97_10` method.
I think you need to just handle it and return `false` in case of `Err`.

Reply:
i agree